docs/pipeline.rst

```diff
@@ -197,3 +197,141 @@ to optimize the conversion by giving the converter additional
 information, even if the model to convert is included in a
 pipeline. That why the option mechanism was implemented:
 :ref:`l-conv-options`.
+
+Investigate discrepencies
+=========================
+
+A wrong converter may introduce introduce discrepencies
+in a converter pipeline but it is not alway easy to
+isolate the source of the differences. The function
+:func:`collect_intermediate_steps
+<skl2onnx.helpers.collect_intermediate_steps>`
+may then be used to investigate each component independently.
+The following piece of code is extracted from unit test
+`test_investigate.py <https://github.com/onnx/sklearn-onnx/
+blob/master/tests/test_investigate.py>`_ and converts
+a pipeline and each of its components independently.
+
+::
+
+    import numpy
+    from numpy.testing import assert_almost_equal
+    from sklearn.pipeline import Pipeline
+    from sklearn.preprocessing import StandardScaler
+    import onnxruntime
+    from skl2onnx.helpers import collect_intermediate_steps, compare_objects
+
+    # Let's fit a model.
+    data = numpy.array([[0, 0], [0, 0], [2, 1], [2, 1]],
+                       dtype=numpy.float32)
+    model = Pipeline([("scaler1", StandardScaler()),
+                      ("scaler2", StandardScaler())])
+    model.fit(data)
+
+    # Convert and collect every operator in a pipeline
+    # and modifies the current pipeline to keep
+    # intermediate inputs and outputs when method
+    # predict or transform is called.
+    steps = collect_intermediate_steps(model, "pipeline",
+                                       [("input",
+                                         FloatTensorType([1, 2]))])
+
+    # Method and transform is called.
+    model.transform(data)
+
+    # Loop on every operator.
+    for step in steps:
```

Comment:
Wording suggestion:
`# Loop over every operator.`

Perhaps also change the for loop wording to `for op in operators` to emphasize that "step" is in fact an ONNX operator?